### PR TITLE
Clarify digest in API specification

### LINKED ISSUE #505: Clarify requirement of Docker-Content-Digest
The V2 API should be serviceable over static web servers. Requiring the `Docker-Content-Digest` header on requests will make these harder to accomplish. From the specification:

> The Docker-Content-Digest header returns the canonical digest of the uploaded blob which may differ from the provided digest. Most clients may ignore the value but if it is used, the client should verify the value against the uploaded blob data.

We need to ensure that the above applies to all uses of `Docker-Content-Digest`, not just the uploads.

Please see https://groups.google.com/forum/#!topic/docker-dev/zyKH0pnyWnI for discussion.

cc @dmcgowan

### LINKED ISSUE #549: Contradictory DELETE documentation in "Docker Registry HTTP API V2"
In the "Docker Registry HTTP API V2" doc (docs/spec/api.md), DELETE has two contradictory documentation instructions.

First in "Deleting an Image": "For deletes, `reference` *must* be a digest or the delete will fail."

In the table shortly lower: "Delete the manifest identified by `name` and `reference` where `reference` can be a tag or digest."

Granted, this will likely be revisited in the course of updating the docs for tags as first class citizens (#173), but it'd be good to make sure to clarify all references consistently.

### PR DESCRIPTION
**Pull request by stevvooe**
Some clarification around manifest deletes have been added, as well.

- [x] Add entry to specification change log

Signed-off-by: Stephen J Day <stephen.day@docker.com>

Closes #505, #549

### CODE REVIEW

**Comment by stevvooe** [comment]
@DreadPirateShawn The API isn't really organized that way. Routes definitions are method agnostic.

**Comment by stevvooe** [comment]
cc @NathanMcCauley

**Comment by dmcgowan** [comment]
LGTM

File: docs/spec/api.md [hunk 1]
```diff
@@ -238,6 +238,84 @@ When a `200 OK` or `401 Unauthorized` response is returned, the
...
+> of tarsum has not been verified. It is recommended that most implementations
+> use sha256 for interoperability.
+
+While the _algorithm_ does allow one to implement a wide variety of
+algorithms, compliant implementations should use sha256. Heavy processing of
+input before calculating a hash is discouraged to avoid degrading the
+uniqueness of the _digest_ but some canonicalization may be performed to
+ensure consistent identifiers.
+
+Let's use a simple example in psuedo-code to demonstrate a digest calculation:
```
Comment (dmp42) [inline docs/spec/api.md]: pseudo

File: docs/spec/api.md [hunk 2]
```diff
@@ -238,6 +238,84 @@ When a `200 OK` or `401 Unauthorized` response is returned, the
...
+let B = sha256(C)
+let D = 'sha256:' + EncodeHex(B)
+let ID(C) = D
+```
+
+Above, we have bytestring _C_ passed into a function, _SHA256_, that returns a
+bytestring B, which is the hash of _C_. _D_ gets the algorithm concatenated
+with the hex encoding of _B_. We then define the identifier of _C_ to _ID(C)_
+as equal to _D_. A digest can be verified by independently calculating _D_ and
+comparing it with idenifier _ID(C)_
```
Comment (dmp42) [inline docs/spec/api.md]: identifier

**Comment by RichardScothern** [comment]
LGTM

**Comment by dmp42** [comment]
Couple of typos - LGTM otherwise!

**Pull request merged**